digitalocean/app_spec.go

```diff
@@ -700,6 +718,68 @@ func appSpecComponentAlerts() *schema.Resource {
 	}
 }
 
+func appSpecLogDestinations() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the log destination",
+			},
+			"papertrail": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: "Papertrail configuration.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"endpoint": {
+							Type:        schema.TypeString,
+							Optional:    true,
```

Comment:
```suggestion
							Required:    true,
```

We can set these to required as that will only apply if the parent is set. So If `papertrail` is used, `endpoint` is required.